server/src/main/java/org/apache/cassandra/sidecar/metrics/CdcMetrics.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.metrics;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import com.codahale.metrics.DefaultSettableGauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricRegistry;
+
+import static org.apache.cassandra.sidecar.metrics.ServerMetrics.SERVER_PREFIX;
+
+/**
+ * Tracks metrics related to cdc functionality provided by Sidecar.
+ */
+public class CdcMetrics
+{
+    public static final String DOMAIN = SERVER_PREFIX + ".Cdc";
+    protected final MetricRegistry metricRegistry;
+    public final NamedMetric<DeltaGauge> cdcRawCleanerFailed;
+    public final NamedMetric<DeltaGauge> orphanedIdx;
+    public final NamedMetric<DefaultSettableGauge<Integer>> oldestSegmentAge;
+    public final NamedMetric<DefaultSettableGauge<Long>> totalConsumedCdcBytes;
```

Comment:
If my understanding is correct, cdc consumption can happen multiple times between the 2 stats publishing internal. In this case, `totalConsumedCdcBytes` should be using `DeltaGauge`. 
The other 2 usages of `DefaultSettableGauge` look valid.